src/traits.rs

```diff
@@ -59,3 +59,18 @@ impl<'a,T> Strided for MutStride<'a,T> {
 impl<'a,T> MutStrided for MutStride<'a,T> {
     fn as_stride_mut(&mut self) -> MutStride<T> { self.reborrow() }
 }
+
+// this isn't as general as it could be.
+impl<'a,T, X: Deref<Target=[T]>> From<&'a X> for Stride<'a,T> {
```

Comment:
It's not automatic, but it is more general, e.g. I believe the current `From` implementation only works with `X = &[T]`, meaning `From::from(&[1, 2, 3])` doesn't work. However, the `AsRef` version should work with `X = [T]` so the latter does work.

(Actually, it will need to be `X: ?Sized + AsRef<[T]>`.)